Mathlib/Topology/UniformSpace/Dini.lean

```diff
@@ -0,0 +1,159 @@
+/-
+Copyright (c) 2024 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import Mathlib.Analysis.Normed.Order.Lattice
+import Mathlib.Topology.ContinuousMap.Ordered
+import Mathlib.Topology.UniformSpace.CompactConvergence
+
+/-! # Dini's Theorem
+
+This file proves Dini's theorem, which states that if `F n` is a monotone increasing sequence of
+continuous real-valued functions on a compact set `s` converging pointwise to a continuous function
+`f`, then `F n` converges uniformly to `f`.
+
+We generalize the codomain from `ℝ` to a normed lattice additive commutative group `G`.
+This theorem is true in a different generality as well: when `G` is a linearly ordered topological
+group with the order topology. This weakens the norm assumption, in exchange for strengthening to
+a linear order. This separate generality is not included in this file, but that generality was
+included in initial drafts of the original
+[PR #19068](https://github.com/leanprover-community/mathlib4/pull/19068) and can be recovered if
+necessary.
+
+The key idea of the proof is to use a particular basis of `𝓝 0` which consists of open sets that
+are somehow monotone in the sense that if `s` is in the basis, and `0 ≤ f ≤ g`, then
+`g ⁻¹' s ⊆ f ⁻¹' s`, and so the proof would work on any topological ordered group possessing
```

Comment:
```suggestion
are somehow monotone in the sense that if `s` is in the basis, and `0 ≤ x ≤ y`, then
`y ∈ s → x ∈ s`, and so the proof would work on any topological ordered group possessing
```
(This is the same thing, but I think mentioning functions makes things less clear)